src/main/java/com/google/sps/data/CalendarAccess.java

```diff
@@ -0,0 +1,91 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.client.util.DateTime;
+import com.google.api.services.calendar.Calendar;
+import com.google.api.services.calendar.CalendarScopes;
+import com.google.api.services.calendar.model.ConferenceData;
+import com.google.api.services.calendar.model.ConferenceSolution;
+import com.google.api.services.calendar.model.ConferenceSolutionKey;
+import com.google.api.services.calendar.model.CreateConferenceRequest;
+import com.google.api.services.calendar.model.Event;
+import com.google.api.services.calendar.model.EventAttendee;
+import com.google.api.services.calendar.model.EventDateTime;
+import com.google.api.services.calendar.model.EventReminder;
+import com.google.api.services.calendar.model.Events;
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.security.GeneralSecurityException;
+import java.util.Collections;
+
+// Handles all things Google Calendar (for now just getting a Meet link).
+public class CalendarAccess {
+  private Calendar service;
+  private static final String CALENDAR_ID = "info@jqed.dev";
+
+  // TODO: remember to write tests in the code that calls CalendarAccess() that handle what happens
+  // per each exception
```

Comment:
If these files already exist but aren't safe for git, remove this TODO.

Reply:
l thought we might make a stub of CalendarAccess that throws the same 4 exceptions.

Reply:
I'd typically recommend dependency injection here so that the constructor does not do non-trivial work. 

```java
public CalendarAccess(Calendar calendar);
```

Then, you can have a `public static` helper function which creates the calendar (and throws all of those exceptions)

```java
public static MakeCalendar(SecretFetcher secretFetcher) {
  String key = secretFetcher.getSecretValue("SERVICE_ACCT_KEY");
  return new Calendar.Builder(
      GoogleNetHttpTransport.newTrustedTransport(),
      JacksonFactory.getDefaultInstance(),
      GoogleCredential.fromStream(new ByteArrayInputStream(key.getBytes()))
            .createScoped(Collections.singletonList(CalendarScopes.CALENDAR)))
    .setApplicationName("Interview Me CalendarAccess")
    .build();
}
```

With the full usage being
```java
CalendarAccess calendarAccess = 
    new CalendarAccess(CalendarAccess.MakeCalendar(
        new SecretFetcher("interview-me-step-2020")));
```

That said, this is usually useful for testing, but I couldn't find a test Calendar class, so I don't know how useful this is. I guess a mock (from mockito) could be used...